shed/schema_test.go

```diff
@@ -124,3 +124,119 @@ func TestDB_schemaIndexPrefix(t *testing.T) {
 		}
 	})
 }
+
+// TestDB_RenameIndex checks if index name is correctly changed.
+func TestDB_RenameIndex(t *testing.T) {
+
+	t.Run("empty names", func(t *testing.T) {
+		db, cleanupFunc := newTestDB(t)
+		defer cleanupFunc()
+
+		// empty names
+		renamed, err := db.RenameIndex("", "")
+		if err == nil {
+			t.Error("error not returned, but expected")
+		}
+		if renamed {
+			t.Fatal("index should not be renamed")
+		}
+
+		// empty index name
+		renamed, err = db.RenameIndex("", "new")
+		if err == nil {
+			t.Error("error not returned, but expected")
+		}
+		if renamed {
+			t.Fatal("index should not be renamed")
+		}
+
+		// empty new index name
+		renamed, err = db.RenameIndex("current", "")
+		if err == nil {
+			t.Error("error not returned, but expected")
+		}
+		if renamed {
+			t.Fatal("index should not be renamed")
+		}
+	})
+
+	t.Run("same names", func(t *testing.T) {
+		db, cleanupFunc := newTestDB(t)
+		defer cleanupFunc()
+
+		renamed, err := db.RenameIndex("index1", "index1")
+		if err != nil {
+			t.Error(err)
+		}
+		if renamed {
+			t.Fatal("index should not be renamed")
+		}
+	})
+
+	t.Run("unknown name", func(t *testing.T) {
+		db, cleanupFunc := newTestDB(t)
+		defer cleanupFunc()
+
+		renamed, err := db.RenameIndex("index1", "index1new")
+		if err != nil {
```

Comment:
is this not supposed to be `if err == nil { t.Fatal("expected error but got none") }`?

Reply:
Current implementation does not return error if the index name is not found, but returns false for renamed.